src/sharedData/components/SharedDataEntry.js

```diff
@@ -0,0 +1,187 @@
+import React from 'react';
+
+import filesize from 'filesize';
+import _ from 'lodash/fp';
+import { useTranslation } from 'react-i18next';
+import { useDispatch, useSelector } from 'react-redux';
+
+import DeleteIcon from '@mui/icons-material/Delete';
+import FileDownloadIcon from '@mui/icons-material/FileDownload';
+import InsertDriveFileOutlinedIcon from '@mui/icons-material/InsertDriveFileOutlined';
+import { Button, Grid, ListItem, Typography } from '@mui/material';
+
+import ConfirmButton from 'common/components/ConfirmButton';
+import EditableText from 'common/components/EditableText';
+import Restricted from 'permissions/components/Restricted';
+
+import { useGroupContext } from 'group/groupContext';
+import { deleteSharedData, updateSharedData } from 'sharedData/sharedDataSlice';
+
+import theme from 'theme';
+
+const ICON_SIZE = 32;
+
+const formatDate = (language, dateString) =>
+  new Intl.DateTimeFormat(language, { dateStyle: 'medium' }).format(
+    Date.parse(dateString)
+  );
+
+const FileIcon = ({ resourceType }) => {
+  switch (resourceType) {
+    case 'csv':
+    case 'xls':
+    case 'xlsx':
+      return (
+        <img
+          style={{ filter: 'opacity(50%)' }}
+          width="24"
+          height="24"
+          src={`/files/${resourceType.substring(0, 3)}.png`}
+          alt={resourceType.toUpperCase()}
+        />
+      );
+    default:
+      return (
+        <InsertDriveFileOutlinedIcon
+          sx={{ fontSize: ICON_SIZE, color: theme.palette.gray.mid2 }}
+        />
+      );
+  }
+};
+
+const SharedDataEntryCard = ({ file }) => {
+  const { i18n, t } = useTranslation();
+  const { group, updateOwner } = useGroupContext();
+  const processing = useSelector(_.get(`sharedData.processing.${file.id}`));
+  const dispatch = useDispatch();
+
+  // TODO: get presigned URL from backend and send user there
```

Comment:
If the `url` attribute of the data entry is being used, so this `TODO` mark is unnecessary.